wherehows-web/app/components/datasets/containers/dataset-health.ts

```diff
@@ -30,6 +50,53 @@ export default class DatasetHealthContainer extends Component {
     // Do something in the future
   });
 
+  actions = {
+    /**
+     * Triggered when the user clicks on one of the bars in the summary charts child component, will trigger
+     * a filter for whatever bar they select, unless it already is one in which case we will remove the filter
+     * @param this - Explicit this declaration for typescript
+     * @param filterDatum - Passed in to the action by the child component, contains the tag to be filtered for
+     */
+    onFilterSelect(this: DatasetHealthContainer, filterDatum: IChartDatum): void {
+      const currentFilters = get(this, 'currentFilters');
+      const filterName = filterDatum.name;
+
+      if (currentFilters.has(filterName)) {
+        currentFilters.delete(filterName);
+      } else {
+        // This strange little logic here makes sure we have only one active filter at a time. Taking this away will
+        // essentially give us multiple filter support (if we decide to make such a UI for it)
+        currentFilters.clear();
+        currentFilters.add(filterName);
+      }
+
+      this.notifyPropertyChange('currentFilters');
+    }
+  };
+
   // Mock data for testing demo purposes, to be deleted once we have actual data and further development
   testSeries = [{ name: 'Test1', value: 10 }, { name: 'Test2', value: 5 }, { name: 'Test3', value: 3 }];
+  fakeCategories: ComputedProperty<Array<IChartDatum>> = computed('currentFilters', function(
```

Comment:
not a fan of pushing code with mock/test data as part of the application, dirties the code and you don't get a clear sense of the boundaries, amongst other things. why not use mirage for these types of things, and keep the actual component clean? You can pass these as component attributes. In the very worst case, the container component can request the data from fixtures
also why not just categories

Reply:
I guess I can get the component to fetch the data from the fixtures, pending future modification. The reason why I wanted to keep it at the component level is that there's no concrete interface with the api even, so I wanted to restrict the data to what it'd look like at the lowest level.